test/unit/storage/LockingResourceStore.test.ts

```diff
@@ -101,4 +95,38 @@ describe('A LockingResourceStore', (): void => {
     expect(lock.release).toHaveBeenCalledTimes(1);
     expect(order).toEqual([ 'acquire', 'release' ]);
   });
+
+  it('releases the lock on the resource when data has been read.', async(): Promise<void> => {
+    const representation = await store.getRepresentation({ path: 'path' }, {});
+    const drainData = new Promise((resolve): any => {
+      representation.data.on('data', (): any => true);
+      representation.data.prependListener('end', (): any => {
+        order.push('end');
+        resolve();
+      });
+    });
+    await drainData;
```

Comment:
I found it hard to find this line in the test, which is where the action happened. Maybe newline above and below?